### PR TITLE
[ GRU ] implement fowarding / backwarding of GRU Layer 

### PR DESCRIPTION
**Pull request by jijoongmoon**
- [ GRU ] implement fowarding / backwarding of GRU Layer
```
This commit includes,
  . forwarding implementation
  . calGradient implementation
  . calDerivative implementation
  . gru_basic unittest

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>
```
- Fix activation type
```
using hidden_state_activation_type

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>
```

### CODE REVIEW

**Review by zhoonit — APPROVED** [review]
LGTM

File: nntrainer/layers/gru.cpp [hunk 1]
```diff
@@ -159,26 +161,219 @@ void GRULayer::setRecurrentActivation(ActivationType activation) {
...
+
+  h_prev.setZero();
+
+  Tensor &hidden_ = hidden->getVariableRef();
+  Tensor &input_ = net_input[0]->getVariableRef();
+
+  Tensor hs_prev;
+  Tensor hs;
+
+  for (unsigned int b = 0; b < input_dim[0].batch(); ++b) {
```
Comment (kparichay) [inline nntrainer/layers/gru.cpp:182]: it might be good to write the formula in comments here
that would also clarify zrg variables.

**Review by kparichay — APPROVED** [review]
LGTM

File: nntrainer/layers/gru.cpp (hunk 1 above)
Comment (jijoongmoon) [reply nntrainer/layers/gru.cpp:187]: done.

**Pull request merged**